spec/shared-examples/subprocess-runner.rb

```diff
@@ -32,25 +32,31 @@
     end
   end
 
-  describe "running 'ls' with a different working directory" do
+  describe "running 'ls' or 'dir' with a different working directory" do
     subject do
-      described_class.new(%w[ls]).tap do |o|
-        o.cwd = fixture_root
-      end
+      if R10K::Util::Platform.windows?
+	      described_class.new(%w[cmd /c dir]).tap do |o|
+          o.cwd = fixture_root
+        end
+	    else
+	      described_class.new(%w[ls]).tap do |o|
+          o.cwd = fixture_root
+        end
+	    end
     end
 
     it "returns the contents of the given working directory" do
       result = subject.run
-      expect(result.stdout).to eq 'no-execute.sh'
+      expect(result.stdout).to match('no-execute.sh')
     end
   end
 
   describe "running 'false'" do
     subject { described_class.new(%w[false]) }
 
-    it "sets the exit code to 1" do
+    it "sets the exit code to 1", unless: R10K::Util::Platform.windows? do
```

Comment:
Why don't we want to test this on windows?

Reply:
false is not a command in powershell like it is in bash, so it doesn't return 1. It ends up being a command not found, so it matches the criteria for 'indicates the command failed'. I don't think it's super useful to test the resulting Powershell response code, since there's not guaranteed like `false` in linux.

Reply:
gotcha. Is there something equivalent to test on Windows? honestly I don't even know what this test is for / why we're testing this at all.. ¯\_(ツ)_/¯

Reply:
Even if with a similar command, the way powershell doesn't return exit codes like bash makes it very different to test for.

I think it's nebulous how exactly we want to support powershell in r10k; it's different enough from bash where there's very limited interoperability, and r10k up to this point doesn't have any explicit support for powershell; we might want invest in more support if we want a larger functionality in windows, but that's way beyond the scope of this PR.